### PR TITLE
remove cache for repo metadata

### PR DESCRIPTION
**Pull request by camdencheek**
Now that we batch requests, performance is quite good on sourcegraph.com
without it, and it does little but add complexity.

For most searches, fetching repo metadata adds less that 15ms of latency, and it doesn't
get much better with the cache since it usually doesn't completely remove the need for a 
db roundtrip, it just makes the payload smaller. 

Additionally, this fixes an issue with repo metadata getting mixed up between repos because
`GetByIDs` doesn't actually return things in the same order it was given.

### CODE REVIEW

File: cmd/frontend/internal/search/search.go [hunk 1]
```diff
@@ -305,20 +302,14 @@ func (h *streamHandler) getEventRepoMetadata(ctx context.Context, event streamin
...
 		}
 
-		var getter repos.MetadataGetter = database.Repos(h.db)
-		if ffs.GetBoolOr("cc_useRepoMetadataCache", false) {
-			getter = repos.CachedMetadataGetter{
-				MetadataGetter: getter,
-				Cache:          h.repoMetadataCache,
-			}
-		}
+		repoMetadata = make(map[api.RepoID]*types.Repo, len(ids))
```
Comment (rvantonder) [inline cmd/frontend/internal/search/search.go:305]: nit: can do `repoMetadata := ...` here and avoid the `var ...` up there no?

**Review by rvantonder — APPROVED** [review]

File: cmd/frontend/internal/search/search.go (hunk 1 above)
Comment (rvantonder) [reply cmd/frontend/internal/search/search.go:305]: oh sorry, I see it's returned, ignore me.

shame-cube

Comment (camdencheek) [reply cmd/frontend/internal/search/search.go]: Changed it to make it less confusing either way 😄

**Pull request merged**